bin/dune_gen.ml

```diff
@@ -32,8 +32,13 @@ let run (`Setup ()) (`Prelude prelude) (`Directories dirs) =
   line "    ]";
   line "  in";
   line "  let predicates = Predicate.[ byte; toploop ] in";
+  line "  let non_deterministic =";
+  line "    match Sys.getenv_opt \"MDX_RUN_NON_DETERMINISTIC\" with";
+  line "    | Some _ -> true";
```

Comment:
I think you want to allow a value that disables, e.g. `MDX_RUN_NON_DETERMINISTIC=0`. You can check how cmdliner uses that.

Reply:
and `getenv_opt` is recent-ish so you might want to check that it's always available when mdx is used, or use `getenv`.

Reply:
According to the [Cmdliner docs](https://erratique.ch/software/cmdliner/doc/Cmdliner/Arg/index.html#val-flag) it doesn't allow it:

> `flag i` is a bool argument defined by an optional flag that may appear at most once on the command line under one of the names specified by `i`. The argument holds true if the flag is present on the command line and false otherwise.

I was surprised myself, but thought it would be better to be consistent between the MDX cli and dune-gen. I can of course also implement disabling the option by un-setting it.

`Sys.getenv_opt` is 4.05+ and if I understand the various lower-bound flags in dune then dune requires 4.08+ so I think we should be fine. But it is a good point that the minium supported OCaml version of MDX needs to match dune's. In our case, it matches it exactly so that's good!

(Also the test compiles the `dune_gen.ml` so from this PR on we are also checking whether the dune-gen output is 4.08+ compatible)

Reply:
all good then!

Reply:
(I don't think we have to follow what dune supports: you can install and run dune on a 4.02 project, but if you use the mdx stanza it's 4.08+)